### PR TITLE
{AKS} Fix #17681: Fix `az aks create` error if name contains characte…

### LINKED ISSUE #17681: 'az aks create' error if name contains character underscore(_)
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

$> az aks create --resource-group wooccdocker -n "aks_woocccluster" --node-count 1 
_--name cannot exceed 63 characters and can only contain letters, numbers, or dashes (-)._

**Expected behavior**

According to the document, underscore can be used for names.
Entity | Scope | Length | Valid Characters
-- | -- | -- | --
managedClusters | resource group | 1-63 | Alphanumerics, underscores, and hyphens.Start and end with alphanumeric.
openShiftManagedClusters | resource group | 1-30 | Alphanumerics.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftcontainerservice

**Environment summary**
$ az --version
azure-cli                         2.22.0

core                              2.22.0
telemetry                          1.0.6

Extensions:
subscription                       0.1.4
blueprint                          0.1.1
account                            0.2.1

### PR DESCRIPTION
**Pull request by RakeshMohanMSFT**
…r underscore

Fixes https://github.com/Azure/azure-cli/issues/17681

**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 1]
```diff
@@ -147,11 +147,11 @@ def validate_linux_host_name(namespace):
     in the CLI pre-flight.
     """
     # https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
-    rfc1123_regex = re.compile(r'^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$')  # pylint:disable=line-too-long
+    rfc1123_regex = re.compile(r'^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-\_]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-\_]{0,61}[a-zA-Z0-9]))*$')  # pylint:disable=line-too-long
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:150]: I checked the regex used in aks-rp, should be `^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$`.
// singleCharacter ^[a-zA-Z0-9]$ or
// more than 2 character with hyphen underscore in middle ^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$
// max length = 63

```suggestion
    rfc1123_regex = re.compile(r'^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$')  # pylint:disable=line-too-long
```

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 2]
```diff
@@ -147,11 +147,11 @@ def validate_linux_host_name(namespace):
...
     found = rfc1123_regex.findall(namespace.name)
     if not found:
-        raise CLIError('--name cannot exceed 63 characters and can only contain '
-                       'letters, numbers, or dashes (-).')
+        raise CLIError('--name can contain only letters, numbers, underscores and hyphens. '
+                       'The name must start and end with letter or number.')
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:154]: Please the info about cannot exceed 63 characters.

**Review by FumingZhang — CHANGES_REQUESTED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

**Pull request merged**